core/src/main/scala/scala/pickling/generator/sourcegen.scala

```diff
@@ -0,0 +1,577 @@
+
+      x.getter match {
+        case x: IrField =>
+          // TODO - handle
+          val tpe = x.tpe[c.universe.type](c.universe)
+          val staticallyElided = tpe.isEffectivelyFinal || tpe.isEffectivelyPrimitive
+          if(x.isScala || !x.isPublic) {
+            // We always have to use reflection for scala fields right now.  Additionally, for Scala fields, we
+            // actually have no idea if they exist at runtime, so we allow failure for now, which is EVIL, but we have no alternative.
+            val result = reflectivelyGet(newTermName("picklee"), x)(fm => fm)
+            val rTerm = c.fresh(newTermName("result"))
+            val logic = q"""val $rTerm = { ..$result }
+                            ${putField(q"$rTerm.asInstanceOf[$tpe]", staticallyElided)}"""
+            if(x.isScala) allowNonExistentField(logic) else logic
+          } else putField(q"picklee.${newTermName(x.fieldName)}", staticallyElided)
+        case _: IrConstructor =>
+          // This is a logic erorr
+          sys.error(s"Pickling logic error.  Found constructor when trying to pickle field ${x.name}.")
+        case y: IrMethod =>
+          val staticallyElided = {
+            // TODO - Figure this out
+            val tpe = y.returnType(c.universe)
+            tpe.isEffectivelyFinal || tpe.isEffectivelyPrimitive
+          }
+          if (y.isPublic) putField(q"picklee.${newTermName(y.methodName)}", staticallyElided)
+          else {
+            val result = reflectivelyGet(newTermName("picklee"), y)(fm => putField(q"${fm}.asInstanceOf[${y.returnType(u).asInstanceOf[c.Type]}]", staticallyElided))
+            q"""..$result"""
+          }
+      }
+    }
+
+
+    def genSubclassDispatch(x: SubclassDispatch): c.Tree = {
+      val tpe = x.parent.tpe[c.universe.type](c.universe)
+      val clazzName = newTermName("clazz")
+        val compileTimeDispatch: List[CaseDef] = (x.subClasses map { subtpe =>
+        val tpe = subtpe.tpe[c.universe.type](c.universe)
+        CaseDef(Bind(clazzName, Ident(nme.WILDCARD)), q"clazz == classOf[$tpe]", createPickler(tpe, q"builder"))
+      })(collection.breakOut)
+
+
+      val failDispatch = {
+        val dispatcheeNames = x.subClasses.map(_.className).mkString(", ")
+        val otherTermName = newTermName("other")
+        val throwUnknownTag =
+          if(x.subClasses.isEmpty) {
+            q"""throw _root_.scala.pickling.PicklingException("Class " + clazz + " not recognized by pickler")"""
+        } else q"""throw _root_.scala.pickling.PicklingException("Class " + clazz + " not recognized by pickler, looking for one of: " + $dispatcheeNames)"""
+        CaseDef(Bind(otherTermName, Ident(nme.WILDCARD)), throwUnknownTag)
+      }
+      val runtimeDispatch = CaseDef(Ident(nme.WILDCARD), EmptyTree, createRuntimePickler(q"builder"))
+      // TODO - Figure out if we can handle runtime dispatch...
+      val unknownDispatch =
+        if(x.lookupRuntime) List(runtimeDispatch)
```

Comment:
This is what happened before for static-only algorithms, so it's no better/worse.

Also, it is actually much safer.